benchmarks/domains/Hardness.py

```diff
@@ -0,0 +1,266 @@
+# Hardness benchmarking, a maximization task on experimental hardness dataset.
+
+from __future__ import annotations
+
+import os
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import scipy as sp
+from pandas import DataFrame
+
+from baybe.campaign import Campaign
+from baybe.parameters import NumericalDiscreteParameter, TaskParameter
+from baybe.recommenders.pure.nonpredictive.sampling import RandomRecommender
+from baybe.searchspace import SearchSpace
+from baybe.simulation import simulate_scenarios
+from baybe.targets import NumericalTarget
+from benchmarks.definition.convergence import (
+    ConvergenceBenchmark,
+    ConvergenceBenchmarkSettings,
+)
+
+# Set up directory and load datasets
+home_dir = os.getcwd()
+# Materials Project (MP) bulk modulus dataset
+df_mp = pd.read_csv(
+    os.path.join(home_dir, "benchmarks", "domains", "mp_bulkModulus_goodOverlap.csv"),
+    index_col=0,
+)
+# Experimental (Exp) hardness dataset
+df_exp = pd.read_csv(
+    os.path.join(home_dir, "benchmarks", "domains", "exp_hardness_goodOverlap.csv"),
+    index_col=0,
+)
+element_cols = df_exp.columns.to_list()[4:]
+
+# Initialize an empty dataframe to store the integrated hardness values
+df_exp_integrated_hardness = pd.DataFrame()
+
+# For each unique composition in df_exp, make a cubic spline interpolation of the hardness vs load curve
+for composition_i in df_exp["composition"].unique():
+    composition_subset = df_exp[df_exp["composition"] == composition_i]
+    # Sort the data by load
+    composition_subset = composition_subset.sort_values(by="load")
+    composition_subset = composition_subset.drop_duplicates(subset="load")
+    if len(composition_subset) < 5:  # Continue to the next composition
+        continue
+
+    # Perform cubic spline interpolation of the hardness vs load curve
+    spline = sp.interpolate.CubicSpline(
+        composition_subset["load"], composition_subset["hardness"]
+    )
+    # Integrate the spline from the minimum load to the maximum load
+    integrated_value = spline.integrate(0.5, 5, extrapolate=True)
+
+    # Make a new dataframe with the element_cols from composition_subset
+    composition_summary = composition_subset[
+        ["strComposition", "composition"] + element_cols
+    ]
+    composition_summary = composition_summary.drop_duplicates(subset="composition")
+    composition_summary["integratedHardness"] = integrated_value
+
+    df_exp_integrated_hardness = pd.concat(
+        [df_exp_integrated_hardness, composition_summary]
+    )
+
+# ----- Target function (integrated hardness) -----

```

Comment:
You have these "headings" all over the place, which is a result of having all logic dumped sequentially in to the main scope of the module. As mentioned in another comment, when you tear apart the logic into individual functions, the need for these headings should naturally disappear